cpp (c++) sample isn't building on Ubuntu 18.04: protobuf error

**Describe the bug**
Can't compile cpp example from samples directory. The wallet-core builds fine but not the cpp sample provided.

**To Reproduce**
To reproduce you can build latest docker image of wallet-core of `master` branch on Ubuntu 18.04 and then try to compile the sample provided in the `samples/cpp` directory.

1.  `git clone https://github.com/trustwallet/wallet-core`
2. `docker build docker/wallet-core`

**Expected behavior**
cpp sample should be built fine as it builds on mac os x.

**Log (snip)**
```
$ docker build docker/wallet-core/Dockerfile
$ docker run -ti 58ec3ce563d1 /bin/bash
root@0146fb6d85a4:/# cd /wallet-core/samples/cpp
root@0146fb6d85a4:/wallet-core/samples/cpp# cmake . -DWALLET_CORE=../../
-- The C compiler identification is Clang 7.0.1
-- The CXX compiler identification is Clang 7.0.1
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
TrustWalletCore library found here: /wallet-core/build/libTrustWalletCore.a
-- Configuring done
-- Generating done
-- Build files have been written to: /wallet-core/samples/cpp

root@0146fb6d85a4:/wallet-core/samples/cpp# make > err 2>&1
root@0146fb6d85a4:/wallet-core/samples/cpp# # head -100 err
Scanning dependencies of target sample
[ 50%] Building CXX object CMakeFiles/sample.dir/sample.cpp.o
[100%] Linking CXX executable sample
CMakeFiles/sample.dir/sample.cpp.o: In function `buildAnySignerInputMsg(unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
sample.cpp:(.text+0x29): undefined reference to `google::protobuf::internal::fixed_address_empty_string[abi:cxx11]'
sample.cpp:(.text+0x40): undefined reference to `google::protobuf::internal::fixed_address_empty_string[abi:cxx11]'
sample.cpp:(.text+0x8f): undefined reference to `google::protobuf::MessageLite::SerializeToString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
CMakeFiles/sample.dir/sample.cpp.o: In function `parseSignedTransactionOutput(void const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)':
sample.cpp:(.text+0x1c3): undefined reference to `google::protobuf::MessageLite::ParseFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Any.pb.cc.o): In function `google::protobuf::internal::ArenaStringPtr::SetNoArena(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)':
/wallet-core/build/protobuf/staging/protobuf-3.9.0/src/google/protobuf/arenastring.h:269: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator=(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Any.pb.cc.o): In function `google::protobuf::internal::ArenaStringPtr::CreateInstanceNoArena(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const*)':
/wallet-core/build/protobuf/staging/protobuf-3.9.0/src/google/protobuf/arenastring.h:379: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Any.pb.cc.o): In function `google::protobuf::internal::ArenaStringPtr::DestroyNoArena(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const*)':
/wallet-core/build/protobuf/staging/protobuf-3.9.0/src/google/protobuf/arenastring.h:325: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Any.pb.cc.o): In function `google::protobuf::internal::ArenaStringPtr::CreateInstance(google::protobuf::Arena*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const*)':
/wallet-core/build/protobuf/staging/protobuf-3.9.0/src/google/protobuf/arena.h:520: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/build/protobuf/staging/protobuf-3.9.0/src/google/protobuf/arena.h:612: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Any.pb.cc.o): In function `void google::protobuf::internal::arena_destruct_object<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(void*)':
/wallet-core/build/protobuf/staging/protobuf-3.9.0/src/google/protobuf/arena.h:96: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `TW::Any::Signer::sign() const':
/wallet-core/src/Any/Signer.cpp:38: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:60: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:60: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:75: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:76: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:77: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:76: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:77: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:89: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o):/wallet-core/src/Any/Signer.cpp:89: more undefined references to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()' follow
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `TW::Any::Signer::sign() const':
/wallet-core/src/Any/Signer.cpp:99: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:100: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:101: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:100: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:101: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:110: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:111: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:112: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:111: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:112: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:124: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o):/wallet-core/src/Any/Signer.cpp:124: more undefined references to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()' follow
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `TW::Any::Signer::sign() const':
/wallet-core/src/Any/Signer.cpp:133: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:134: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:135: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:134: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:135: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:154: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:155: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:156: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:155: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:156: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:165: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:166: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:167: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:166: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:167: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:178: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o):/wallet-core/src/Any/Signer.cpp:178: more undefined references to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()' follow
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `TW::Any::Signer::sign() const':
/wallet-core/src/Any/Signer.cpp:199: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/wallet-core/src/Any/Signer.cpp:200: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:201: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:200: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:201: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Any/Signer.cpp:217: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o):/wallet-core/src/Any/Signer.cpp:217: more undefined references to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()' follow
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string<decltype(nullptr)>(char const*)':
/usr/bin/../include/c++/v1/string:813: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__init(char const*, unsigned long)'
/usr/bin/../include/c++/v1/string:813: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(char const*, unsigned long)':
/usr/bin/../include/c++/v1/string:1799: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__init(char const*, unsigned long)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `TW::Cosmos::Proto::SigningInput::set_private_key(void const*, unsigned long)':
/wallet-core/src/Cosmos/../proto/Cosmos.pb.h:3743: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Cosmos/../proto/Cosmos.pb.h:3743: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(char const*, unsigned long)':
/usr/bin/../include/c++/v1/string:1799: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__init(char const*, unsigned long)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `TW::Binance::Proto::SigningInput::set_private_key(void const*, unsigned long)':
/wallet-core/src/Binance/../proto/Binance.pb.h:6290: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Binance/../proto/Binance.pb.h:6290: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > TW::hex<std::__1::__wrap_iter<unsigned char*> >(std::__1::__wrap_iter<unsigned char*>, std::__1::__wrap_iter<unsigned char*>)':
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:28: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::reserve(unsigned long)'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:32: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::push_back(char)'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:33: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::push_back(char)'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:37: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:37: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(char const*, unsigned long)':
/usr/bin/../include/c++/v1/string:1799: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__init(char const*, unsigned long)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `TW::Ethereum::Proto::SigningInput::set_private_key(void const*, unsigned long)':
/wallet-core/src/Wanchain/../proto/Ethereum.pb.h:876: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Wanchain/../proto/Ethereum.pb.h:876: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(Signer.cpp.o): In function `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > TW::hex<std::__1::__wrap_iter<char*> >(std::__1::__wrap_iter<char*>, std::__1::__wrap_iter<char*>)':
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:28: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::reserve(unsigned long)'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:32: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::push_back(char)'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:33: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::push_back(char)'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:37: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
/wallet-core/src/Bitcoin/../Zcash/../HexCoding.h:37: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::~basic_string()'
```

--- a/samples/cpp/CMakeLists.txt
+++ b/samples/cpp/CMakeLists.txt
@@ -55,7 +55,6 @@ if (APPLE)
 	set (PLATFORM_LINK_FLAGS "-framework Foundation -framework OpenCL")
 elseif (WIN32)
 	set (PLATFORM_LINK_FLAGS "/STACK:8000000") #provides 8MB default stack size for linux on windows
-else ()
 endif ()
 
 SET (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${PLATFORM_LINK_FLAGS}")

--- a/samples/cpp/CMakeLists.txt
+++ b/samples/cpp/CMakeLists.txt
@@ -39,6 +39,8 @@ else ()
 	add_compile_options (-Werror=switch)
 endif ()
 
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+
 set (CMAKE_C_STANDARD 11)
 set (CMAKE_C_STANDARD_REQUIRED ON)
 
@@ -54,7 +56,6 @@ if (APPLE)
 elseif (WIN32)
 	set (PLATFORM_LINK_FLAGS "/STACK:8000000") #provides 8MB default stack size for linux on windows
 else ()
-	set (PLATFORM_LINK_FLAGS "-static-libgcc -static-libstdc++")
 endif ()
 
 SET (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${PLATFORM_LINK_FLAGS}")

